### PR TITLE
adding pypi upload

### PR DESCRIPTION
**Pull request by jfriel-oqc**

### CODE REVIEW

File: .github/workflows/package-publish.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install poetry build
+      run: poetry install
+
+    - name: Build a binary wheel and a source tarball
+      run: poetry build
```
Comment (hamidelmaazouz) [inline .github/workflows/package-publish.yml:24]: what about minimal qat compiler build ?

File: .github/workflows/package-publish.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    - uses: actions/checkout@v4
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.x"
+    - name: Install poetry build
```
Comment (hamidelmaazouz) [inline .github/workflows/package-publish.yml:20]: minor: rename to "Poetry build"

File: .github/workflows/package-publish.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        GITHUB_TOKEN: ${{ github.token }}
+      # Upload to GitHub Release using the `gh` CLI.
+      # `dist/` contains the built packages, and the
+      # sigstore-produced signatures and certificates.
+      run: >-
+        gh release upload
+        '${{ github.ref_name }}' dist/**
+        --repo '${{ github.repository }}'
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
```
Comment (hamidelmaazouz) [inline .github/workflows/package-publish.yml:99]: this is the third "Install Poetry" step, wondering if we cache the poetry env 🤔

File: .github/workflows/package-publish.yml (hunk 1 above)
Comment (jfriel-oqc) [reply .github/workflows/package-publish.yml:24]: doing that after this, just want the core mechanism in place first

File: .github/workflows/package-publish.yml (hunk 3 above)
Comment (jfriel-oqc) [reply .github/workflows/package-publish.yml:99]: I can remove these instances actually, we only ever build once and don't want to cache anything here I don't think. Should always be a fresh build

**Review by helmaazouz — APPROVED** [review]

**Review by hamidelmaazouz — APPROVED** [review]

**Comment by hamidelmaazouz** [comment]
Added https://github.com/oqc-community/qat/issues/136 for the CI failure. I've been seeing this intermittently recently. Needs investigation, but it's unrelated to this PR

**Comment by jfriel-oqc** [comment]
> Added #136 for the CI failure. I've been seeing this intermittently recently. Needs investigation, but it's unrelated to this PR

I'm pretty sure it's just the runner running out of memory every now and then, but will check it out

**Comment by jfriel-oqc** [comment]
failing pipe because the release already exists, I'm going to merge this

**Pull request merged**